packages/avatars/src/styled/utility.ts

```diff
@@ -16,40 +16,41 @@ const [xxs, xs, s, m, l] = SIZE;
 export const TRANSITION_DURATION = 0.25;
 
 export interface IStyledStatusIndicatorProps extends ThemeProps<DefaultTheme> {
-  readonly size?: IAvatarProps['size'];
-  readonly type?: IAvatarProps['status'] | 'active';
+  readonly $size?: IAvatarProps['size'];
+  readonly $type?: IAvatarProps['status'] | 'active';
 }
 
+const StatusColorParams = {
+  active: { hue: 'crimson', light: { shade: 700 }, dark: { shade: 600 } },
+  available: { hue: 'mint', light: { shade: 500 }, dark: { shade: 400 } },
+  away: { hue: 'orange', light: { shade: 500 }, dark: { shade: 400 } },
+  transfers: { hue: 'azure', light: { shade: 500 }, dark: { shade: 400 } },
+  offline: { hue: 'grey', light: { shade: 500 }, dark: { shade: 400 } }
+};
+
 export function getStatusColor(
-  type?: IStyledStatusIndicatorProps['type'],
-  theme?: IStyledStatusIndicatorProps['theme']
+  theme: IStyledStatusIndicatorProps['theme'],
+  type?: IStyledStatusIndicatorProps['$type']
 ): string {
-  switch (type) {
-    case 'active':
-      return getColorV8('crimson', 400, theme)!;
-    case 'available':
-      return getColorV8('mint', 400, theme)!;
-    case 'away':
-      return getColorV8('orange', 400, theme)!;
-    case 'transfers':
-      return getColorV8('azure', 400, theme)!;
-    case 'offline':
-      return getColorV8('grey', 500, theme)!;
-    default:
-      return 'transparent';
+  if (type === undefined) {
+    return 'transparent';
   }
+
+  const colorArgs = StatusColorParams[type];
+
+  return getColor({ ...colorArgs, theme });
```

Comment:
```suggestion
  return getColor({ ...colorArgs, theme }) || 'transparent';
```

The only thing is that there might be an unprotected path where a consumer provides a rogue status value :shrug:. Might not be worth sealing off.

Reply:
You beat me to it. Just pushed that up. I did something similar, but I think both could be valid. I'm OK switching to you what here if you think that's worth it.